MechJeb2/MechJebModuleNodeExecutor.cs

```diff
@@ -223,89 +199,95 @@ public override void OnFixedUpdate()
+        {
+            BurnTime(_dvLeft, out double halfBurnTime, out double spool);
+            if (_isLoadedPrincipia)
+                // in principia node.UT is the start of the burn and we need to subtract off the spool time
+                return _hasNodes ? node.UT - spool - LeadTime : -1;
+
+            // in stock node.UT is the center of the burn and the halfBurnTime calculation has the spool time
+            return node.UT - halfBurnTime - LeadTime;
+        }
+
         private void DecrementDvLeft()
         {
             if (!_isLoadedPrincipia) return;
 
-            if ((BurnTriggered || _nearingBurn) && MuUtils.PhysicsRunning())
-            {
-                // decrement remaining dV based on engine and RCS thrust
-                // Since this is Principia, we can't rely on the node's delta V itself updating, we have to do it ourselves.
-                // We also can't just use vesselState.currentThrustAccel because only engines are counted.
-                // NOTE: This *will* include acceleration from decouplers, which is pretty cool.
-                Vector3d dV = (Vessel.acceleration_immediate - Vessel.graviticAcceleration) * TimeWarp.fixedDeltaTime;
-                _dvLeft -= Vector3d.Dot(dV, Core.Attitude.targetAttitude());
-            }
+            if (!BurnTriggered || !MuUtils.PhysicsRunning()) return;
+
+            // decrement remaining dV based on engine and RCS thrust
+            // Since this is Principia, we can't rely on the node's delta V itself updating, we have to do it ourselves.
+            // We also can't just use vesselState.currentThrustAccel because only engines are counted.
+            // NOTE: This *will* include acceleration from decouplers, which is pretty cool.
+            Vector3d dV = (Vessel.acceleration_immediate - Vessel.graviticAcceleration) * TimeWarp.fixedDeltaTime;
+            _dvLeft -= Vector3d.Dot(dV, Core.Attitude.targetAttitude());
         }
 
         private void HandleAutowarp(double ignitionUT)
         {
-            double timeToBurn = ignitionUT - VesselState.time - LeadTime;
-            //autowarp, but only if we're already aligned with the node
-            if (Autowarp && !BurnTriggered)
+            if (BurnTriggered || !Autowarp) return;
+
+            double timeToBurn = ignitionUT - VesselState.time;
+
+            if (timeToBurn > 600)
             {
-                if (timeToBurn > 600 || (MuUtils.PhysicsRunning()
-                        ? Core.Attitude.attitudeAngleFromTarget() < 1 && Core.vessel.angularVelocity.magnitude < 0.001
-                        : Core.Attitude.attitudeAngleFromTarget() < 10))
-                {
-                    Core.Warp.WarpToUT(VesselState.time + timeToBurn);
-                }
-                else
-                {
-                    //realign
-                    Core.Warp.MinimumWarp();
-                }
+                Core.Warp.WarpToUT(ignitionUT - 600);
```

Comment:
I don't think this is correct; I think you want to check attitude still. In this case we'll gradually step down warp as we get close to 600s from node, then (if we started the warp aligned) step it back up again once we pass < 600s from node (!).

You want to warp all the way until the lead time if you're already aligned; you only want to break at ~600s if you're not aligned already.